client/blocks/login/index.jsx

```diff
@@ -46,6 +47,12 @@ class Login extends Component {
 		}
 	};
 
+	componentWillReceiveProps = ( nextProps ) => {
+		if ( this.props.twoFactorAuthType !== nextProps.twoFactorAuthType ) {
```

Comment:
Good catch @yurynix, I didn't think at all about `requestError`. Looking at the props I don't think we need to handle anything else but let me know if I missed something.